docs: release notes for 1.23.2, 1.22.4, and 1.21.6 (EOL) (#4832)
cloudnative-pg/cloudnative-pg: docs/src/release_notes/v1.21.md

[commits](https://github.com/cloudnative-pg/cloudnative-pg/commits/release-1.21)
on the release branch in GitHub.

## Version 1.21.6

> **Release date:** Jun 12, 2024

Reviewer: note to self: this may be a GH diff artifact, but seems the 1.21.5 has been deleted

!!! Warning
    This is expected to be the last release in the 1.21.X series.
    Users are encouraged to update to a newer minor version soon.

### Enhancements:

> - Enabled configuration of standby-sensitive parameters during recovery from a
>   physical backup (#4564)

Reviewer: The phrasing is a bit odd: "recovery from a physical backup", looks like the
physical backup is a failure.
```suggestion
- Enabled configuration of standby-sensitive parameters during recovery using a
  physical backup (#4564)
```

- Enabled the configuration of the liveness probe timeout via the
  `.spec.livenessProbeTimeout` option (#4719)

- `cnpg` plugin for `kubectl`:

>     - Introduced support for ANSI colors in the plugin through the `--color`

Reviewer: We already had support for colors, we're adding configurability.
```suggestion
    - Enhanced support for ANSI colors in the plugin by adding the `--color`
```

      option, which accepts `always`, `never`, and `auto` (default) as values
      (#4775)
    - The plugin is now available on Homebrew for macOS users (#4602)

> 

Reviewer: This PR is only backported to 1.23, no 1.21 nor 1.22. Do we still want it in the RN since the plugin is built on 1.23?

Reviewer 2: Homebrew always delivers the latest version, and that version is retrocompatible. I would keep it.

### Fixes:

- Prevented fenced instances from entering an unnecessary loop and consuming
  all available CPU (#4625)

- Resolved an issue where the instance manager on the primary would
  indefinitely wait for the instance to start after encountering a failure
  following a stop operation (#4434)

> - Fixed an issue where the interaction between `hotstandby_feedback` and

Reviewer 3: ```suggestion
- Fixed an issue where the interaction between `hot_standby_feedback` and
```

>   managed cluster-level replication slots was preventing the removal of
>   obsolete replication slots; this issue was causing disk space to remain
>   occupied by dead tuples (#4811)

Reviewer: This is not my understanding of the issue.
```suggestion
  managed cluster-level replication slots was wrongly setting xmin on
  standby slots, which was then set on the primary and prevented VACUUM
  from removing dead tuples and freeing disk space (#4811)
```

Reviewer 2: I think both versions are wrong. It's not true that `hot_standby_feedbak` was preventing the removal of obsolete replication slot, not it sets the `xmin` on standby. It only propagates the `xmin` from the standby (where it was harmless) to the primary. I would not go into details and rater say:

```suggestion
  managed cluster-level replication slots was preventing the autovacuum from operate correctly; this issue was causing disk space to remain
  occupied by dead tuples (#4811)
```

Reviewer: s/from operate/from operating/
and, line seems very long, otherwise 👍 on that, Marco.

Reviewer 2: updated

- Fixed a panic in the backup controller that occurred when pod container
  statuses were missing (#4765)

- Prevented unnecessary shutdown of the instance manager (#4670)

> - Prevented unnecessary reloads of PostgreSQL triggered by `SIGHUP` issued by
>   the instance manager (#4531)

Reviewer 2: The `SIGHUP` is the symptom, not the cause. I would say: 

```suggestion
- Prevented unnecessary reloads of PostgreSQL configuration when there are no changes (#4531)
```

- Prevented unnecessary reloads of the ident map by ensuring a consistent and
  unique method of writing its content (#4648)

- Avoided conflicts during phase registration by patching the status of the
  resource instead of updating it (#4637)

- Implemented a timeout when restarting PostgreSQL and lifting fencing (#4504)

- Ensured that a replica cluster is restarted after promotion to properly set
  the archive mode (#4399)

> - Removed a stale keep-alive connection with volume snapshot backups that was
>   causing random failures (#4768)

Reviewer 2: I would say:

```suggestion
- Removed unneeded concurrent keep-alive routine that was causing random failures
  in volume snapshot backups (#4768)
```
